src/ros_filter.cpp

```diff
@@ -771,21 +771,20 @@ namespace RobotLocalization
       else
       {
         ROS_WARN_STREAM("use_control is set to true, but acceleration_limits is missing. Will use default values.");
-        accelerationLimits.resize(TWIST_SIZE, 1.0);
       }
 
-      accelerationGains.resize(TWIST_SIZE, 1.0);
       if(nhLocal_.getParam("acceleration_gains", accelerationGains))
       {
-        if(accelerationGains.size() != TWIST_SIZE)
+        const size_t size = accelerationGains.size();
+        if(size != TWIST_SIZE)
         {
           ROS_ERROR_STREAM("Acceleration gain configuration must be of size " << TWIST_SIZE <<
-            ". Provided config was of size " << accelerationGains.size() << ". All gains will be assumed to be 1.");
+            ". Provided config was of size " << size << ". All gains will be assumed to be 1.");
+          std::fill_n(accelerationGains.begin(), size, 1.0);
           accelerationGains.resize(TWIST_SIZE, 1.0);
         }
       }
 
-      bool useAccelLimits = false;
```

Comment:
As I mentioned on the description, if `deceleration_limits` is given, we don't set `useAccelLimits` to `true`. Either way, IMHO it's more consistent with the rest of the logic to remove this variable and always set `decelerationGains = accelerationGains` when the `deceleration_gains` param is omitted.